Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionForcePointFirewall.yaml

```diff
@@ -0,0 +1,473 @@
+    ];
+    let parser = (disabled:bool) 
+    { 
+      CommonSecurityLog
+            | where not(disabled)
+            | where DeviceFacility !in ("Management","Authentication") // Add Filters here for Network Session data. Right now removing the known Audit events only
+            | where DeviceVendor=="FORCEPOINT" and DeviceProduct=="Firewall"
+            | where isnotempty(Message)
+            | lookup ActionLookup on DeviceAction
+            | lookup ProtocolLookup on Protocol
+            | lookup SubTypeLookup on DeviceEventClassID
+            | lookup EventResultMapping on Message
+            | extend DvcAction = coalesce(DvcAction,DvcAction1, DvcAction2), 
+                     EventResult = coalesce(EventResult, EventResult1, EventResult2), 
+                     EventSeverity = coalesce(EventSeverity, EventSeverity1)
+            | project-away DvcAction1, EventSeverity1, EventResult1, DvcAction2, EventResult2
+            | lookup ApplicationProtocolLookup on ApplicationProtocol
+            | extend 
+                    EventCount = toint(1),
+                    EventType = "NetworkSession",
+                    EventSchema = "NetworkSession",
+                    EventSchemaVersion = "0.2.6"
+            | parse-kv AdditionalExtensions as (
+                        requestURL:string
+                    ) with (pair_delimiter=';', kv_delimiter='=')
+            | project-rename
+                    EventVendor = DeviceVendor,
+                    EventProduct = DeviceProduct,
+                    EventOriginalType = DeviceEventClassID,
+                    DstPortNumber = DestinationPort,
+                    DstIpAddr = DestinationIP,
+                    SrcPortNumber = SourcePort,
+                    SrcIpAddr = SourceIP,
+                    DstNatIpAddr = DestinationTranslatedAddress,
+                    DstNatPortNumber = DestinationTranslatedPort,
+                    SrcNatIpAddr = SourceTranslatedAddress,
+                    SrcNatPortNumber = SourceTranslatedPort,
+                    EventProductVersion = DeviceVersion,
+                    EventMessage = Message,
+                    DvcOriginalAction = DeviceAction,
+                    SrcBytes = SentBytes,
+                    DstBytes = ReceivedBytes,
+                    EventOriginalSubType = DeviceFacility,
+                    DvcId = DeviceExternalID,
+                    DvcInboundInterface = DeviceInboundInterface,
+                    DvcOutboundInterface = DeviceOutboundInterface,
+                    DvcIpAddr = DeviceAddress,
+                    EventOriginalSeverity = LogSeverity,
+                    DvcDomain = Computer        
+            | extend
+                    EventStartTime = todatetime(ReceiptTime),
+                    EventEndTime = todatetime(ReceiptTime),
+                    boolHasIP = iif(indexof_regex(requestURL,@"(\d+.\d+.\d+.\d+)") == -1, false, true),
```

Comment:
Just to recap what we discussed in our meeting on this topic.
The requestURL will always be a domain name and not the complete URL. It can sometimes be the resolved IP address as well which is same as DestinationIP as far as I have seen. 
Apart from requestMethod (provided sometimes), there is no other WebSession information provided like User Agent, Status Code or Content Type. Thus we decided to stick with NetworkSession only. Do you think we have to reconsider this?
Also, same is true for File and DNS events as well. The event is marked in the respective category but no relevant information is given. Should we change the EventType in that case?